src/abstractdataframe/abstractdataframe.jl

```diff
@@ -197,17 +193,23 @@ julia> rename!(uppercase, df)
 function rename!(df::AbstractDataFrame, vals::AbstractVector{Symbol};
                  makeunique::Bool=false)
     rename!(index(df), vals, makeunique=makeunique)
+    # renaming columns of SubDataFrame has to clean non-note metadata in its parent
+    _drop_all_nonnote_metadata!(parent(df))
```

Comment:
This is general. If we mutate `AbstractDataFrame` we need to drop `:note` metadata from its `parent` as it gets invalidated.

Reply:
Maybe that's a more general question: shouldn't we preserve column-level metadata of unaffected columns, whatever its style? I can't see a case where they could get invalidated by renaming other columns.

Reply:
This was my major problem with `:none` style metadata. I had to decide on some rule when it is dropped.
I proposed: "drop it always" as this is the simplest rule and most conservative one.
Another rule could be: "drop it only if the column is affected". The problem is that:
1. still some users might say that changing the name of column `"a"` invalidated metadata for column `"b"` because it e.g. referred to column `"a"` also.
2. We will open a rabbit-hole of different cases when it is allowed to keep `:none`-style meatadata even if dataframe is transformed.

Reply:
I find the expectation 1. quite convoluted. :-) The only situation I can imagine is if some metadata said "this column was computed as the sum of columns x, y and z". It doesn't sound too dangerous if that kind of thing gets out of sync with the other columns. Anyway we have noted that rules are supposed to work in the most common case and that people who have special needs should note use `:note`.

2 is indeed a general risk, but do you think things would really get worse? We already had to define relatively complex rules, and `transform(df, :x => :y)` preserves metadata for unaffected columns. Wouldn't it be consistent to do the same when columns are renamed using other functions?

Reply:
We have clarified with @nalimilan what what we do now is OK, so I resolve this.